src/devices/Mcp3428/ConversionResult.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Mcp3428::Mcp3428
+// Filename: ConversionResult.cs
+// Created: __
+// Edited: 20190405
+// Creator: Máté Kullai
+
+using System;
+
+namespace Iot.Device.Mcp3428
+{
+    public interface IAdcResult
```

Comment:
this and ConversionResultHandler don't seem to be used anywhere here. I'd recommend removing the interface unless there is a good reason to have it. For ADC `Func<double>` should be enough to abstract it

Reply:
Would be nice to have some common ADC interfaces for the library.

Reply:
@kmate95 - I agree and I've already tried to add one in the past but got outvoted that we should just use `Func<double>` instead of the interface since that's what ADC produces and value doesn't have any specific unit so it's not suitable for the interface

Reply:
I'm not sure I agree with this reasoning. Every ADC unit on every uC **always** produces a voltage value. That voltage may have other meaning like current measurement, weight, pressure, anything.   
That is the concern of the consuming software. Yet the ADC will have an intermediate voltage value to convert. Like this 4 channel converter, the first channel can be temperature, the second humidity, the third pot-meter and the forth a rocket launch countdown. But they always have to go through a (channel number)=>(voltage)=>(anything) interface. And if the first hop is abstracted away, only the business relevant part is left to the consumer.